src/Access/UsersConfigAccessStorage.cpp

```diff
@@ -52,6 +53,38 @@ namespace
 
     UUID generateID(const IAccessEntity & entity) { return generateID(entity.getType(), entity.getName()); }
 
+    template <typename T>
+    void parseGrant(T & entity, const String & string_query)
+    {
+        ParserGrantQuery parser;
+        parser.parseWithoutGrantees();
+
+        String error_message;
+        const char * pos = string_query.data();
+        auto ast = tryParseQuery(parser, pos, pos + string_query.size(), error_message, false, "", false, 0, 0);
+
+        if (!ast)
+            throw Exception(ErrorCodes::BAD_ARGUMENTS, "Failed to parse grant query. Error: {}", error_message);
+
+        auto & query = ast->as<ASTGrantQuery &>();
+
+        if (query.roles)
+            throw Exception(ErrorCodes::BAD_ARGUMENTS, "Roles can't be granted in config file");
```

Comment:
Roles defined in the config file could be granted to other roles defined in the config file.

Reply:
Yes, it's possible. But it will require some refactoring to implement (because you need to know the role's uuid to grant it to another role), and I wonder if it is even so useful.

Reply:
This isn't very difficult - see `allowed_parent_profile_ids` in `parseSettingsProfile()` for how profiles can be referenced in other profiles.

Reply:
Yes, you were right, I missed it!